modules/gis/src/lib/reproject.js

```diff
@@ -0,0 +1,36 @@
+export function mapBinaryCoords(binaryFeatures, fn) {
+  // Expect binaryFeatures to have points, lines, and polygons keys
+  for (const binaryFeature of Object.values(binaryFeatures)) {
+    const {positions} = binaryFeature;
+    for (let i = 0; i < positions.value.length; i += positions.size) {
+      positions.value.set(fn(Array.from(positions.value.subarray(i, i + positions.size))), i);
```

Comment:
Classic case of too much going on here on this line, split on multiple line with self-documenting temporary variable name(s) to keep the intermediate results.